src/csharp/Microsoft.Spark.Worker/Command/SqlCommandExecutor.cs

```diff
@@ -165,6 +165,11 @@ private void WriteOutput(Stream stream, IEnumerable<object> rows, int sizeHint)
             if (s_outputBuffer == null)
                 s_outputBuffer = new byte[sizeHint];
 
+            if (rows.FirstOrDefault() is GenericRow)
+            {
+                rows = rows.Select(r => (object)(r as GenericRow).Values).AsEnumerable();
```

Comment:
Can we double check we are not creating a separate list? cc: @suhsteve

Reply:
From what I observed is that `rows` become `Null` after this statement. `.Show()` will work but not `.Collect()`

Reply:
did you look into why?

Reply:
UdfDf.Collect() returns Null, will look more into GetRows function.

Reply:
ah ok. then it's not this code here. Can you print out the schema? We may need to update pickler, and we should do it as a separate PR since it's covering an independent issue.

Can you try to repro this without udf? You can look at the schema where it fails and try to repro it with json data file.

Reply:
I can repro this without udf via reading json data file.

I have captured the exception and changed the test accordingly. Thanks!